api/src/opentrons/protocol_api/core/protocol_api/legacy_module_core.py

```diff
@@ -87,12 +87,18 @@ def get_deck_slot(self) -> DeckSlotName:
         """Get the module's deck slot."""
         return DeckSlotName.from_primitive(self._geometry.parent)  # type: ignore[arg-type]
 
+    def get_display_name(self) -> str:
+        """Get the module's display name."""
+        return f"{self.get_model()} on {self.get_deck_slot()}"
```

Comment:
`get_model` is different than "display name". Let's tweak `ModuleGeometry` a little so we can do this:

```suggestion
        return self._geometry.display_name
```

Where `ModuleGeometry.display_name` is set to the already existing `__init__` arg called `display_name`. This will require a slight tweak to move some logic in `ModuleGeometry.__init__` to `ModuleGeometry.__repr__`, which we can walk through together if you like